### PR TITLE
NDCubeSequence plotting mixin API.

### PR DESCRIPTION
**Pull request by DanRyanIrish**
This PR introduces a new API for an ```NDCubeSequence``` plotting that will provide a more comprehensive plotting functionality for ```NDCubeSequence``` and more consistent API.  This API/functionality should be mirrored ```NDCube``` where appropriate and vice versa.

This new mixin API will also allow a new ```NDCubeSequenceBase``` class independent of visualization and also of ```sunpy```.

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
@Cadair, I think this is ready for a serious discussion now.

As an explanation, the ```cubesequence``` arg is there just so I can easily play with it without actually mixing in the class.  All occurrences will be removed and replaced with ```self``` before merging.

**Comment by DanRyanIrish** [comment]
I think I might have done another Danny's-1st-PR-style move here.  +1,198 lines, -3 lines...woops!

File: ndcube/mixins/sequence_plotting.py [hunk 1]
```diff
@@ -0,0 +1,1198 @@
...
+            and other dimensions represented by animations sliders.
+            Default=[-1, -2].  If sequence only has one dimension (or effectively one if
+            plot_as_cube is True), plot_axis_indices is ignored and a staice 1D line plot
+            is produced.
+
+        axes_coordinates: `list` of physical coordinates for image axes and sliders or `None`
+            If None coordinates derived from the WCS objects will be used for all axes.
+            If a list, it should contain one element for each axis.  Each element should
+            be either an `astropy.units.Quantity` or a `numpy.ndarray` of coordinates for
+            each pixel, or a `str` denoting a valid extra coordinate.
```
Comment (Cadair) [inline ndcube/mixins/sequence_plotting.py]: add `None` here

File: ndcube/mixins/sequence_plotting.py [hunk 2]
```diff
@@ -0,0 +1,1198 @@
...
+
+        axes_units: length 2 `list` of `astropy.unit.Unit` or valid unit `str` or None.
+            Denotes unit in which X-axis and Y-axis, respectively, should be displayed.
+            Only used if corresponding entry in axes_coordinates is a Quantity.
```
Comment (DanRyanIrish) [inline ndcube/mixins/sequence_plotting.py]: Error in this kwarg description.  Should be same length as number of dimensions.

However, should probably also accept length 2 for images and length one for line plots so slider units don't have to be changed every time for high dimension sequences.

File: ndcube/mixins/sequence_plotting.py (hunk 1 above)
Comment (DanRyanIrish) [inline ndcube/mixins/sequence_plotting.py]: Error in this kwarg description.  Elements can also be None.

axes_coordinates should probably also accept length 2 for images and length one for line plots so slider units don't have to be changed every time for high dimension sequences.

File: ndcube/mixins/sequence_plotting.py [hunk 3]
```diff
@@ -0,0 +1,1198 @@
...
+
+        data_unit: `astropy.unit.Unit` or valid unit `str` or None
+            Unit in which data in a 2D image or animation should be displayed.  Only used if
+            visualization is a 2D image or animation, i.e. if image_axis has length 2.
```
Comment (DanRyanIrish) [inline ndcube/mixins/sequence_plotting.py]: Correct this description.  Data unit also is applied to y-axis of line plot.

**Comment by DanRyanIrish** [comment]
Move the plot_as_cube options into separate ```plot_as_cube``` top level method.

**Comment by DanRyanIrish** [comment]
Move some helper functions into class.  Those that use the object itself.

**Comment by DanRyanIrish** [comment]
Wrap dimension-specific prep operations in ```plot``` into dimension-specific helper functions to clean up ```plot```.

**Comment by DanRyanIrish** [comment]
As part of this PR I've realised there are shortcomings in the 2D images produced, it that coordinates for an axis are sometimes taken from the first sub-cube in the sequence and assumed not to change.  I haven't fully figured out all the ties this occurs.  So this comment is just a reminder to raise an issue after this PR is merged.  Addressing this issue is beyond the scope of this PR.

**Comment by DanRyanIrish** [comment]
Hey @Cadair.  I'm planning to merge this tomorrow (your today) as I think it's ready and there are a couple PRs piling up waiting for it.  I know you're really busy so if you don't have time to look at it no problem.  Otherwise I'll take any comments you do have into tomorrow.

**Pull request merged**